### PR TITLE
feat: add provisioners view to organization settings

### LINKED ISSUE #14436: UI: Organization Provisioners Page
On this page a user should be able to:

- [ ] Understand that they are looking at provisioners grouped by authentication types
- [ ] View which provisioners are using a specific provisioner key
- [ ] View which provisioners are using the PSK
- [ ] View which provisioners are built-in
- [ ] View details about each provisioner such as scope and version (PSK tags and other details will be handled in a separate ticket)
- [ ] View details about each provisioner key such as name, tags, created at, last used
- [ ] CTA to create a new provisioner (links to docs)
- [ ] Add storybooks

Relies on backend work - https://github.com/coder/coder/issues/14439

### PR DESCRIPTION
**Pull request by aslilac**
Closes #14436

- [x] View details about each provisioner such as scope and version (PSK tags and other details will be handled in a separate ticket)
- [x] View details about each provisioner such as name, tags, created at, last used
- [ ] View provisioner warnings
- [x] CTA to create a new provisioner (links to docs)
- [x] Add storybooks

Warnings aren't implemented yet on the backend, but the rest of this should be ready to go! I can handle those as part of another PR to avoid blocking this work if they're not done soon.

### CODE REVIEW

**Comment by jaaydenh** [comment]
Both organization provisioners and groups are enterprise features, however groups displays the enterprise banner but for provisioners an error message is displayed. It would make sense for both of these cases to have the same behavior and show the `Paywall` component.

Screenshot 2024-09-04 at 2 42 02 PM

**Comment by jaaydenh** [comment]
Asking this mainly because I want to update the docs and want to fully understand the terminology. How did you decide that Provisioner.tsx and ProvisionerTag.tsx are modules and something like OrganizationAutocomplete or Paywall is a component? Should OrganizationAutocomplete or Paywall actually have been modules?

**Comment by aslilac** [comment]
I think `OrganizationAutocomplete` and `Paywall` should actually both be modules as well. I think "components" are atomic, simple, and not tied to our specific product; they should be useful in other settings as well.

**Comment by jaaydenh** [comment]
Nice, I like the distinction that Components could be technically be used in any project and still make sense.

File: site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+						endIcon={<OpenInNewIcon />}
+						href={docs("/admin/provisioners")}
+					>
+						Create a provisioner
+					</Button>
+				}
+			>
+				<PageHeaderTitle>Provisioners</PageHeaderTitle>
+			</PageHeader>
+			<Stack spacing={4.5}>
```
Comment (jaaydenh) [inline site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.tsx:33]: I feel the way this is defined `gap: spacing * 8,` sort of constrains the UI to consistent spacing in multiples of 8 if we only use whole numbers for spacing. I would argue that for consistency we should keep whole numbers or rethink the spacing * 8 definition.

Comment (aslilac) [reply site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.tsx:33]: I think we should rethink the `* 8` definition. If you do a search for `spacing={` you'll see that the codebase is already covered in `.5` spacings. 😅 This is just the spacing that the health page already uses, and I think it looks right.

File: site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	return (
+		<div>
+			<PageHeader
+				// The deployment settings layout already has padding.
+				css={{ paddingTop: 0 }}
+				actions={
+					<Button
+						endIcon={<OpenInNewIcon />}
+						href={docs("/admin/provisioners")}
+					>
```
Comment (jaaydenh) [inline site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.tsx:26]:
```suggestion
					<Button
						endIcon={<OpenInNewIcon />}
						target="_blank"
						href={docs("/admin/provisioners")}
					>
```
Its nice when docs open in a new page to take the user out of their current flow.

**Review by jaaydenh — APPROVED** [review]
LGTM

**Pull request merged**